File: worker/src/monitor.rs
use reqwest::Client;
use std::time::{Duration, Instant};

pub async fn check_website(client: &Client, url: &str) -> (bool, u128) {

    let start = Instant::now();

    let res = client
        .get(url)
        .timeout(Duration::from_secs(5))
        .send()
        .await;

    let latency = start.elapsed().as_millis();

    match res {
        Ok(r) => (r.status().is_success(), latency),
        Err(_) => (false, latency),
    }
}

File: client/src/lib/monitor.ts
export const checkWebsite = async (url: string) => {
  const start = Date.now()

  try {
    await fetch(url, { mode: "no-cors" })
    const latency = Date.now() - start

    return {
      status: "UP",
      latency,
    }
  } catch {
    return {
      status: "DOWN",
      latency: 0,
    }
  }
}